client/src/components/PortfolioTable.tsx

```diff
@@ -0,0 +1,741 @@
+              size: "auto",
+            },
+          ],
+        },
+        {
+          header: i18n._(t`Information`),
+          footer: (props) => props.column.id,
+          columns: [
+            {
+              accessorKey: "yearbuilt",
+              header: i18n._(t`Built`),
+              cell: (info) => info.getValue(),
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+            },
+            {
+              accessorKey: "unitsres",
+              header: i18n._(t`Units`),
+              cell: (info) => info.getValue(),
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+              filterFn: "inNumberRange",
+            },
+          ],
+        },
+        {
+          header: i18n._(t`RS Units`),
+          footer: (props) => props.column.id,
+          columns: [
+            {
+              accessorKey: "rsunits2007",
+              header: "2007",
+              cell: (info) => info.getValue(),
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+            },
+            {
+              accessorKey: "rsunitslatest",
+              header: rsunitslatestyear,
+              cell: ({ row }) => {
+                return (
+                  <span
+                    className={`${
+                      // TODO: double check this works with nulls
+                      row.original.rsunitslatest! < row.original.rsunits2007! ?? false
+                        ? "text-danger"
+                        : ""
+                    }`}
+                  >
+                    {row.original.rsunitslatest}
+                  </span>
+                );
+              },
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+              filterFn: "isNonZero",
+            },
+          ],
+        },
+        {
+          header: i18n._(t`HPD Complaints`),
+          footer: (props) => props.column.id,
+          columns: [
+            {
+              accessorKey: "totalcomplaints",
+              header: i18n._(t`Total`),
+              cell: (info) => info.getValue(),
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+            },
+            {
+              accessorKey: "recentcomplaints",
+              header: i18n._(t`Last 3 Years`),
+              cell: (info) => info.getValue(),
+              footer: (props) => props.column.id,
+              enableColumnFilter: false,
+              size: "auto",
+            },
+            {
+              accessorKey: "recentcomplaintsbytype",
+              header: i18n._(t`Top Complaint`),
+              cell: ({ row }) => {
+                const mostCommonType = findMostCommonType(row.original.recentcomplaintsbytype);
```

Comment:
i'm not super familiar with React.memo, does that ensure that this function is only being called once per row vs on render?

Reply:
I'm also not super familiar, but that was my understanding as well. I saw that this was always used in the tanstack table examples